Hangman/HangmanGame.py

```diff
@@ -0,0 +1,119 @@
+from Hangman import Hangman
+from WordGenerator import WordGenerator
+
+
+class HangmanGame:
+    def __init__(self) -> None:
+        self.total_points = 0
+        self.word_generator = WordGenerator()
+        self.hangman = Hangman()
+        self.word = ''
+        self.incorrect_guess_count = 0
+        self.word_display = []
+        self.guesses = []
+        self.status = ''
+
+    def start_game(self) -> None:
+        self.__set_word(self.word_generator.get_next_word())
+        self.__set_status()
+        self.word_display = ['_' for _ in self.word]
+        print("Welcome to hangman")
+        print(self.get_position())
+
+        game_over = (self.get_status() == 'won' or self.get_status() == 'lost')
+
+        # Take input till game is not over
+        while not game_over:
+            self.guess_letter()
+            print(self.get_hangman())
+            print(self.get_position())
+            game_over = (self.get_status() == 'won' or self.get_status() == 'lost')
+
+        if self.get_status() == 'won':
+            print('Congratulations! You guessed the word correctly.')
+        else:
+            print(f'The word is: {self.get_word()}')
+            print('I am sure you will do better next time. :)')
+
+        want_to_restart = (input("To restart, enter Yes: (case insensitive) ")).lower()
+        if want_to_restart == "yes":
+            self.__init__()
```

Comment:
we are caling the constructor inside the object method itself ? in this case it will work but in general you __init__ does a lot more, ideally there either should be a reset function or a class function which creates/manages multiple objects and multiple play by calling object.single_play()